src/types.ts

```diff
@@ -1,9 +1,12 @@
-export type fetcherFn<Data> = (...args: any) => Data | Promise<Data>
-export interface ConfigInterface<
+// Internal types
+
+export type Fetcher<Data> = (...args: any) => Data | Promise<Data>
+
+export type Configuration<
   Data = any,
   Error = any,
-  Fn extends fetcherFn<Data> = fetcherFn<Data>
-> {
+  Fn extends Fetcher<Data> = Fetcher<Data>
+> = {
```

Comment:
is this `=` by purpose?

Reply:
Yeah I changed it from interface to type